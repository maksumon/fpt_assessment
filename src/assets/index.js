export const ESSENTIAL_HEALTH_SCREENING = require('./images/essential_health_screening.jpeg');
export const ADVANCE_HEALTH_SCREENING = require('./images/advance_health_screening.jpeg');
export const NO_SUGAR = require('./images/no_sugar.jpg');
export const FIVE_K_RUN_WALK = require('./images/5k_run_walk.jpeg');
export const EIGHT_GLASSES = require('./images/8_glasses.jpeg');
