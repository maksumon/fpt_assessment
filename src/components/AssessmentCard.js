import React from 'react';
import {View, StyleSheet, Image} from 'react-native';
import {Text, Card} from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';

export const AssessmentCard = ({item}) => {
  return item === null || item === undefined ? (
    <View />
  ) : (
    <View style={styles.container}>
      <Card containerStyle={styles.cardContainer}>
        <View style={styles.row}>
          <Image style={styles.image} resizeMode="cover" source={item.image} />
          <View style={styles.column}>
            <Text style={styles.title}>{item.title}</Text>
            <View style={[styles.row, styles.flexEnd]}>
              <View style={styles.column}>
                <Text style={styles.general}>Earn up to</Text>
                <Text style={styles.general}>
                  <Text style={styles.point}>{item.points} </Text>
                  pts
                </Text>
              </View>
              <Icon
                name="arrow-right"
                size={15}
                color="#900"
                style={styles.arrowRight}
              />
            </View>
          </View>
        </View>
      </Card>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  cardContainer: {
    width: 260,
    marginStart: 2,
    shadowColor: '#000000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  row: {
    flexDirection: 'row',
  },
  column: {
    flexDirection: 'column',
    flexShrink: 1,
    justifyContent: 'space-between',
  },
  image: {
    width: 64,
    height: 96,
    marginHorizontal: 10,
    marginVertical: 5,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 16,
    marginTop: 5,
  },
  general: {
    fontSize: 14,
    marginTop: 5,
  },
  point: {
    fontSize: 14,
    color: '#87ceeb',
    marginTop: 5,
  },
  flexEnd: {
    alignItems: 'flex-end',
  },
  arrowRight: {
    position: 'absolute',
    bottom: 0,
    right: 0,
  },
});
