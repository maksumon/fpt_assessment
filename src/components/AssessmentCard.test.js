import React from 'react';
import renderer from 'react-test-renderer';
import {AssessmentCard} from './AssessmentCard';
import {ADVANCE_HEALTH_SCREENING} from '../assets';

it('renders correctly when there is no item', () => {
  const tree = renderer.create(<AssessmentCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there 99999 points', () => {
  const hugePoints = {
    title: 'Advanced Health Screening',
    points: 99999,
    image: ADVANCE_HEALTH_SCREENING,
  };
  const tree = renderer.create(<AssessmentCard item={hugePoints} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is no point in the item', () => {
  const noPoint = {
    title: 'Advanced Health Screening',
    image: ADVANCE_HEALTH_SCREENING,
  };
  const tree = renderer.create(<AssessmentCard item={noPoint} />).toJSON();
  expect(tree).toMatchSnapshot();
});
