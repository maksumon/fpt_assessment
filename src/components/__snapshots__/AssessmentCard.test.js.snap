// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there 99999 points 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "elevation": 5,
        "margin": 15,
        "marginBottom": 0,
        "marginStart": 2,
        "padding": 15,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": 260,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/images/advance_health_screening.jpeg",
            }
          }
          style={
            Object {
              "height": 96,
              "marginHorizontal": 10,
              "marginVertical": 5,
              "width": 64,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "column",
              "flexShrink": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#242424",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginTop": 5,
              }
            }
          >
            Advanced Health Screening
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "flex-end",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#242424",
                    "fontSize": 14,
                    "marginTop": 5,
                  }
                }
              >
                Earn up to
              </Text>
              <Text
                style={
                  Object {
                    "color": "#242424",
                    "fontSize": 14,
                    "marginTop": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#87ceeb",
                      "fontSize": 14,
                      "marginTop": 5,
                    }
                  }
                >
                  99999
                   
                </Text>
                pts
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#900",
                    "fontSize": 15,
                  },
                  Object {
                    "bottom": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when there is no item 1`] = `<View />`;

exports[`renders correctly when there is no point in the item 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#e1e8ee",
        "borderWidth": 1,
        "elevation": 5,
        "margin": 15,
        "marginBottom": 0,
        "marginStart": 2,
        "padding": 15,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 3.84,
        "width": 260,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "transparent",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/images/advance_health_screening.jpeg",
            }
          }
          style={
            Object {
              "height": 96,
              "marginHorizontal": 10,
              "marginVertical": 5,
              "width": 64,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "column",
              "flexShrink": 1,
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#242424",
                "fontSize": 16,
                "fontWeight": "bold",
                "marginTop": 5,
              }
            }
          >
            Advanced Health Screening
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "alignItems": "flex-end",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "flexShrink": 1,
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#242424",
                    "fontSize": 14,
                    "marginTop": 5,
                  }
                }
              >
                Earn up to
              </Text>
              <Text
                style={
                  Object {
                    "color": "#242424",
                    "fontSize": 14,
                    "marginTop": 5,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#87ceeb",
                      "fontSize": 14,
                      "marginTop": 5,
                    }
                  }
                >
                   
                </Text>
                pts
              </Text>
            </View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#900",
                    "fontSize": 15,
                  },
                  Object {
                    "bottom": 0,
                    "position": "absolute",
                    "right": 0,
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
